packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -763,18 +763,133 @@ String _debugCompareFloats(String labelA, double valueA, String labelB, double v
          'apply the min() or max() functions, or the clamp() method, to the $labelB? ';
 }
 
-// ///
-// /// ## Writing a RenderSliver subclass
-// ///
-// /// ### Painting
-// ///
-// /// The [paint] method is called with an [Offset] to the top-left corner of the
-// /// sliver, _regardless of the axis direction_.
-// ///
-// /// ### childScrollOffset
-// ///
-// /// If the subclass positions children anywhere other than at scroll offset
-// /// 0.0, you need to override [childScrollOffset]...
+/// Base class for the render objects that implement scroll effects in viewports.
+///
+/// A [RenderViewport] has a list of child slivers. Each sliver — literally a
+/// thin piece of the viewport's contents — is laid out in turn, so as to cover
+/// the viewport.
+///
+/// Slivers participate in the _sliver protocol_, which is essentially a 15+
```

Comment:
added:
```
/// (The "15+ dimensions" refers
/// to the number of properties found on [SliverConstraints] and
/// [SliverGeometry]. The number is not precise because some of the properties
/// are interdependent.)
```